starknet_py/utils/data_transformer/data_transformer.py

```diff
@@ -30,6 +31,38 @@
 CairoData = List[int]
 
 
+def construct_result_object(result: dict) -> NamedTuple:
+    fields = result.keys()
+    res_tuple = namedtuple(
+        field_names=fields,
+        typename="Result",
+        rename=True,
+    )
+    # pylint: disable=protected-access
+    mappings = dict(zip(fields, res_tuple._fields))
+    value = {mappings[key]: value for key, value in result.items()}
+    tpl_value = res_tuple(**value)
```

Comment:
Maybe some changes in naming? I'm not sure if it's important here, but it would be great to have more descriptive ones.